app/src/main/java/com/keylesspalace/tusky/view/MediaPreviewImageView.kt

```diff
@@ -15,45 +15,67 @@
 package com.keylesspalace.tusky.view
 
 import android.content.Context
+import android.graphics.Matrix
 import android.support.v7.widget.AppCompatImageView
 import android.util.AttributeSet
+import com.keylesspalace.tusky.entity.Attachment
 
-import com.keylesspalace.tusky.util.FocalPointEnforcer
+import com.keylesspalace.tusky.util.FocalPointUtil
+import com.squareup.picasso.Callback
 
 /**
- * This is an extension of the standard android ImageView, which makes sure to call a focal point
- * enforcer when its size changes.
+ * This is an extension of the standard android ImageView, which makes sure to update the custom
+ * matrix when its size changes if a focal point is set.
  *
- * If the focalPointEnforcer is set on this view, it will call this enforcer to update the image
+ * If a focal point is set on this view, it will use the FocalPointUtil to update the image
  * matrix each time the size of the view is changed. This is needed to ensure that the correct
  * cropping is maintained.
  *
- * However if there is no focalPointEnforcer set (e.g. it is null), then this view should simply
+ * However if there is no focal point set (e.g. it is null), then this view should simply
  * act exactly the same as an ordinary android ImageView.
  */
 class MediaPreviewImageView
 @JvmOverloads constructor(
 context: Context,
 attrs: AttributeSet? = null,
 defStyleAttr: Int = 0
-) : AppCompatImageView(context, attrs, defStyleAttr) {
-    private var focalPointEnforcer: FocalPointEnforcer? = null
+) : AppCompatImageView(context, attrs, defStyleAttr), Callback {
+    private var focus: Attachment.Focus? = null
+    private var focalMatrix: Matrix? = null
 
     /**
-     * Set the focal point enforcer for this view.
+     * Set the focal point for this view.
      */
-    fun setFocalPointEnforcer(enforcer: FocalPointEnforcer) {
-        this.focalPointEnforcer = enforcer
+    fun setFocalPoint(focus: Attachment.Focus) {
+        scaleType = ScaleType.MATRIX
+        this.focus = focus
+        this.focalMatrix = Matrix()
```

Comment:
I would allocate it only if it's not already there

Reply:
Good point, I've changed it :+1: